pkg/config/validation/validation.go

```diff
@@ -214,6 +214,21 @@ func ValidateHTTPHeaderName(name string) error {
 	return nil
 }
 
+// ValidateStringMatch validates a StringMatch
+func ValidateStringMatch(m *networking.StringMatch) error {
+	if m == nil {
+		return nil
+	}
+	switch x := m.MatchType.(type) {
+	case *networking.StringMatch_Regex:
+		// Default max size for safe regex is 100
+		if len(x.Regex) > 100 {
```

Comment:
Can we explicitly specify the max size for safe regex somewere, and reuse the same constant here?